lib/tokenlist.h

```diff
@@ -217,6 +245,10 @@ class CPPCHECKLIB TokenList {
     /** File is known to be C/C++ code */
     bool mIsC;
     bool mIsCpp;
+
+    std::function<void()> mValuesFunc;
+    mutable bool mValuesSet;
+    mutable bool mInSetValues;
```

Comment:
These two variables can be moved into the `mValuesFunc`.

Reply:
We could do something similar to what we do for [`memoize`](https://github.com/danmar/cppcheck/blob/main/lib/astutils.cpp#L2564):

```cpp
template<class F>
static std::function<void()> once(F f)
{
    bool ran = false;
    return [=]() mutable {
        if (ran)
            return;
        ran = true;
        f();
        return result;
    };
}
```

Reply:
This was very hacky. I just tried to get rid what I wanted it to do.

> These two variables can be moved into the `mValuesFunc`.

Good point.

> We could do something similar to what we do for [`memoize`]

Will take a look. I know there are patterns for this but I rarely use them so I keep forgetting them.